src/XRP/default-xrp-client.ts

```diff
@@ -435,9 +435,12 @@ export default class DefaultXrpClient implements XrpClientDecorator {
    * @see https://xrpl.org/depositauth.html
    *
    * @param wallet The wallet associated with the XRPL account enabling Deposit Authorization and that will sign the request.
-   * @returns A promise which resolves to a string representing the hash of the submitted AccountSet transaction.
+   * @returns A promise which resolves to a Tuple<string, TransactionStatus> representing the hash of the submitted AccountSet transaction
+   *          and the status of the transaction.
    */
-  public async enableDepositAuth(wallet: Wallet): Promise<string> {
+  public async enableDepositAuth(
+    wallet: Wallet,
+  ): Promise<[string, TransactionStatus]> {
```

Comment:
I'd prefer this API to be an object (`Transaction Result`?) rather than an array. 

As a non-JS native programmer, arrays of heterogenous objects don't make a ton of sense to me 
 and objects can have names that confer meaning rather than just indices. It's easier to say something like:

```js
const result = enableDepositAuth(...) 
console.log(result.status)
```

than
```js
console.log(result[1])
```

Reply:
cool, definitely agree with this and was considering creating an object